src/Microsoft.Extensions.SecretManager.Tools/Program.cs

```diff
@@ -147,27 +143,80 @@ internal int RunInternal(params string[] args)
 
         internal string ResolveIdFromProject(string projectPath)
         {
-            var finder = new GracefulProjectFinder(_workingDirectory);
+            var finder = new MsBuildProjectFinder(_workingDirectory);
             var projectFile = finder.FindMsBuildProject(projectPath);
 
             Logger.LogDebug(Resources.Message_Project_File_Path, projectFile);
 
+            var targetFile = GetTargetFile();
+            var outputFile = Path.GetTempFileName();
+
             try
             {
-                var project = new MsBuildProjectContextBuilder()
-                    .UseMsBuild(_msBuildContext)
-                    .AsDesignTimeBuild()
-                    .WithBuildTargets(Array.Empty<string>())
-                    .WithProjectFile(projectFile)
-                    .WithTargetFramework("") // TFM doesn't matter
-                    .Build();
-
-                return project.GetUserSecretsId();
+                var commandResult = Command.CreateDotNet("msbuild",
```

Comment:
In theory, that could work, but I'm not planning on anything like that unless we can get a design from CLI folks that offers clear benefits.